src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/jpa/entity/BookingEntity.kt

```diff
@@ -128,15 +129,32 @@ interface BookingRepository : JpaRepository<BookingEntity, UUID> {
   )
   fun findByBedIdAndArrivingBeforeDate(bedId: UUID, date: LocalDate, thisEntityId: UUID?): List<BookingEntity>
 
+  /*
+    This query is to find the closest booking to the start date for the current bedspace search
+    The ClosestBooking is to get the closest booking to the bedspace search start date that is not cancelled
+  */
   @Query(
-    "SELECT b FROM BookingEntity b WHERE (b.bed, b.departureDate) IN (" +
-      "  SELECT b2.bed, MAX(b2.departureDate)" +
-      "  FROM BookingEntity b2 " +
-      "  WHERE b2.departureDate <= :date " +
-      "  AND b2.bed.id IN :bedIds " +
-      "  AND SIZE(b2.cancellations) = 0 " +
-      "  GROUP BY b2.bed " +
-      ")",
+    """
+      WITH ClosestBooking AS
+         (SELECT b.bed_id, MAX(b.departure_date) departure_date
+          FROM bookings b
+          LEFT JOIN cancellations c ON b.id = c.booking_id
+          WHERE b.departure_date <= :date
+            AND b.bed_id IN :bedIds
+            AND c.id IS NULL
+          GROUP BY b.bed_id)
+          
+      SELECT b.*,
```

Comment:
Do we want to be more specific with the columns pulled through on the booking?
I am wondering if we need to pull through the application, as that then has a oneToMany relationship with assessment. Maybe we could be really specific here - or maybe go down the chain of other things being pulled in from other models and lazy load other oneToMany relationships where relevant (such as assessments in the applicationEntity)

Just a thought

Reply:
No because it maps the query result to BookingEntity